### ISSUE TITLE
Luma errors when using Vite: Super expression must either be null or a function

### ISSUE THREAD

**Issue by amcdnl** [visgl/luma.gl#1659]
# Issue
I'm migrating from CRA to Vite for my UI builds, however, when I try to run the project I get the following error:

```
inherits.js:4 Uncaught TypeError: Super expression must either be null or a function
```

which traces back to:

![image](https://user-images.githubusercontent.com/227909/185615966-6c911fb3-8142-494b-9c32-2103669ae3a9.png)

if I look at the code throwing the error:

![image](https://user-images.githubusercontent.com/227909/185620899-3f31838d-2077-4f4c-8720-d81ee34a276c.png)

It shows that `superClass` is `null` which is referring to the `Resource` class.

If you do some quick research, it seems like its related to the class extends with static props: https://stackoverflow.com/questions/30116430/reactjs-giving-error-uncaught-typeerror-super-expression-must-either-be-null-or

## Versions
```
"@msrvida/sanddance-react": "3.0.1",
"@deck.gl/core": "8.3.7",
"@deck.gl/layers": "8.3.7",
"@luma.gl/core": "8.3.1",
"vite": "^3.0.8",
```

**Comment by ibgreen** [comment]
On the master branch all examples in the luma.gl examples folder use vite and this has not come up. It could of course be specific to the v8 release branch. An isolated repro case might be useful.

**Comment by Pessimistress** [comment]
There could have been some circular referencing that was fixed between 8.3 and 8.5.

My suggestion is to upgrade deck/luma to the latest releases. There are plenty of users using vite/esbuild right now.

**Comment by Pessimistress** [comment]
@danmarshall
